helidon-cli/impl/src/main/java/io/helidon/build/cli/impl/DevCommand.java

```diff
@@ -13,52 +13,64 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.helidon.build.cli.impl;
 
-import java.util.Properties;
+package io.helidon.build.cli.impl;
 
 import io.helidon.build.cli.harness.Command;
 import io.helidon.build.cli.harness.CommandContext;
 import io.helidon.build.cli.harness.CommandExecution;
 import io.helidon.build.cli.harness.Creator;
 import io.helidon.build.cli.harness.Option.Flag;
+import io.helidon.build.util.ProjectConfig;
 
 /**
  * The {@code dev} command.
  */
 @Command(name = "dev", description = "Continuous application development")
 public final class DevCommand extends BaseCommand implements CommandExecution {
 
-    private static final String MAVEN_PLUGIN = "maven.plugin";
+    private static final String CLEAN_PROP_PREFIX = "-Ddev.clean=";
+    private static final String FORK_PROP_PREFIX = "-Ddev.fork=";
+    private static final String MAVEN_PLUGIN_PROPERTY = "maven.plugin";
+    private static final String DEV_GOAL = "dev";
+    private static final String DEFAULT_DEV_GOAL = "helidon:" + DEV_GOAL;
 
     private final CommonOptions commonOptions;
     private final boolean clean;
     private final boolean fork;
 
     @Creator
     DevCommand(
-            CommonOptions commonOptions,
-            @Flag(name = "clean", description = "Perform a clean before the first build") boolean clean,
-            @Flag(name = "fork", description = "Fork mvn execution") boolean fork) {
+        CommonOptions commonOptions,
+        @Flag(name = "clean", description = "Perform a clean before the first build") boolean clean,
+        @Flag(name = "fork", description = "Fork mvn execution") boolean fork) {
         this.commonOptions = commonOptions;
         this.clean = clean;
         this.fork = fork;
     }
 
     @Override
     public void execute(CommandContext context) {
-        Properties cliConfig = cliConfig();
-        String cleanProp = "-Ddev.clean=" + clean;
-        String forkProp = "-Ddev.fork=" + fork;
-        StringBuilder pluginGoal = new StringBuilder(cliConfig.getProperty(MAVEN_PLUGIN));
-        pluginGoal.append(":")
-                .append(helidonVersion())
-                .append(":dev");
+        final ProjectConfig projectConfig = projectConfig(commonOptions.project().toPath());
```

Comment:
Minor comment. The use of final for local vars is generally inconsistent, we should drop it like we decided in Helidon.

Reply:
I thought we only decided not to use `final` on parameters...?

Reply:
My understanding is that we decided _only_ for fields. Looking at Helidon code, most local vars are not final.